lib/ansible/modules/cloud/amazon/cloudformation.py

```diff
@@ -297,6 +305,32 @@ def create_stack(module, stack_params, cfn):
     return result
 
 
+def create_changeset(module, stack_params, cfn):
+    if 'TemplateBody' not in stack_params and 'TemplateURL' not in stack_params:
+        module.fail_json(msg="Either 'template' or 'template_url' is required.")
+
+    try:
+        if not 'ChangeSetName' in stack_params:
+            # Determine ChangeSetName using hash of parameters, if anyone has a better idea I'm all ears.
+            changeset_name = 'Ansible-' + stack_params['StackName'] + '-' + secure_hash_s(json.dumps(stack_params, sort_keys=True))
```

Comment:
Wouldn't this create a conflict if someone ran this task twice with the same params, thus resulting in the same name (and a conflict)? Generating the name this way seems good, but can you add a check to make sure there isn't a changeset with that name already? 

If there is one, then you could return unchanged because the changeset already exists so this task's work is already done.